packages/cli/src/lib/vite/worker-entry.ts

```diff
@@ -0,0 +1,198 @@
+/**
+ * All the code in this file is executed in workerd. This file
+ * is compiled at build time (tsup) to be transpiled to JS, and
+ * then loaded as string in a workerd instance at runtime as
+ * the worker entrypoint. It then requests modules to Vite
+ * and evaluates them to run the app's backend code.
+ */
+
+import {
+  ViteRuntime,
+  ssrModuleExportsKey,
+  type ViteRuntimeModuleContext,
+} from 'vite/runtime';
+import type {HMRPayload} from 'vite';
+import type {Response} from 'miniflare';
+
+export interface ViteEnv {
+  __VITE_ROOT: string;
+  __VITE_HMR_URL: string;
+  __VITE_FETCH_MODULE_PATHNAME: string;
+  __VITE_RUNTIME_EXECUTE_URL: string;
+  __VITE_WARMUP_PATHNAME: string;
+  __VITE_SETUP_ENV: (request: Request) => void;
+  // Ref: https://github.com/cloudflare/workerd/blob/main/src/workerd/api/unsafe.h
+  __VITE_UNSAFE_EVAL: {
+    eval(code: string, name?: string): Function;
+    newFunction(code: string, name?: string, ...args: string[]): Function;
+    newAsyncFunction(code: string, name?: string, ...args: string[]): Function;
+  };
+}
+
+const O2_PREFIX = '[o2:runtime]';
+
+export default {
+  /**
+   * Worker entry module that wraps the user app's entry module.
+   */
+  async fetch(request: Request, env: ViteEnv, ctx: any) {
+    env.__VITE_SETUP_ENV(request);
```

Comment:
What does this method do?

Reply:
It modifies `globalThis` with an utility for the Subrequest Profiler. Once Vite is stable, we can deprecate the code we have in `remix-oxygen` for this and move it here.

To see more details around how this works, check [`wrappedBindings` in the `mini-oxygen.ts` file](https://github.com/Shopify/hydrogen/blob/b85203b6c047de62bb9394718da30de92dbdbbe3/packages/cli/src/lib/vite/mini-oxygen.ts#L80-L95). The [`setupScripts` part comes from the Hydrogen plugin](https://github.com/Shopify/hydrogen/blob/b85203b6c047de62bb9394718da30de92dbdbbe3/packages/cli/src/lib/vite/hydrogen-middleware.ts#L16).